### PR TITLE
Added an important & informative book for Algorithm and Data Structures using Javascript. 

### PR DESCRIPTION
**Pull request by RahulBRB**
## What does this PR do?
Add resource(s) 

### Description
Added new informative book for learning Algorithm and Data structures using JavaScript, written by Loiane Groner.

### Why is this valuable (or not)?
It is valuable because this book is designed to teach DSA concepts using JavaScript. It covers a wide range of data structures and algorithms, including arrays, linked lists, stacks, queues, trees, sorting and searching algorithms, and more, all implemented in JavaScript.

### How do we know it's really free?
Can read it directly from the provided link.

### For book lists, is it a book? For course lists, is it a course? etc.
Yes, it's a book

## Checklist:
- [x ] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x ] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x ] Include author(s) and platform where appropriate.
- [x ] Put lists in alphabetical order, correct spacing.
- [x ] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-subjects.md [hunk 1]
```diff
@@ -112,6 +112,7 @@ Books that cover a specific programming language can be found in the [BY PROGRAM
 * [Essential Algorithms](https://www.programming-books.io/essential/algorithms/) - Krzysztof Kowalczyk and  Stack Overflow Documentation project (HTML)
 * [Foundations of Computer Science](http://infolab.stanford.edu/~ullman/focs.html) - Al Aho, Jeff Ullman
 * [Lectures Notes on Algorithm Analysis and Computational Complexity (Fourth Edition)](https://ianparberry.com/books/free/license.html) - Ian Parberry (use form at bottom of license)
+* [Learning JavaScript Data Structures and Algorithms](https://github.com/amilajack/reading/blob/master/JavaScript/Learning%20Javascript%20Data%20Structures%20and%20Algorithms.pdf) - Loiane Groner (PDF)
```
Comment (mohdahsanrazakhan) [inline books/free-programming-books-subjects.md:115]: Swap line number `114` with `115` for alphabetical ordering.

**Review by mohdahsanrazakhan — COMMENTED** [review]
Improve the code to fix the `linter error`

**Review by RahulBRB — COMMENTED** [review]
Swapped line 114 with 115

**Comment by eshellman** [comment]
This appears not to be an authorized distribution. Please see docs/CONTRIBUTING.md guidelines.

**Pull request closed without merging**